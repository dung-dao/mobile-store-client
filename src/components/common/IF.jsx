import Proptypes from "prop-types";

const IF = (props) => {
  if (props.condt)
    return props.children;
  else
    return null;
};

IF.propTypes = {
  condt: Proptypes.bool.isRequired,
};

export default IF;
