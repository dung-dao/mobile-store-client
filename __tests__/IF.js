import React from 'react';
import renderer from 'react-test-renderer';
import IF from "../src/components/common/IF";


describe('IF children', () => {
    it('should be null when condition is false', function () {
        const emptyTree = renderer.create(<IF condt={false}>a</IF>).toJSON();
        expect(emptyTree).toBe(null);
    });

    it('Should match snapshot', () => {
        const tree = renderer.create(<IF condt={true}>a</IF>).toJSON();
        expect(tree).toMatchSnapshot()
    });
})


