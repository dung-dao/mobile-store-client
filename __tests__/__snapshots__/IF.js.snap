// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IF children Should match snapshot 1`] = `"a"`;
